extensions/git/src/git.ts

```diff
@@ -520,6 +520,13 @@ export class Git {
 			}
 		}
 
+		// Handle symbolic links
+		// Git 2.31 added --path-format flag to rev-parse which allows us to get the relative path of the repository root
+		if (!pathEquals(repositoryPath, repoPath) && !isDescendant(repoPath, repositoryPath) && !isDescendant(repositoryPath, repoPath) && this.compareGitVersionTo('2.31.0') !== -1) {
```

Comment:
In order to make this more readable, should we rename `repositoryPath` to `pathInsidePossibleRepository` and `repoPath` to `repositoryRootPath`?